fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2068,5 +2076,33 @@ public static void analyzeExpressionIgnoreSlot(Expr expression, ConnectContext s
                 new Scope(RelationId.anonymous(), new RelationFields()));
     }
 
+    public static class CheckUserVariableDependencyVisitor implements AstVisitor<Void, Scope> {
+        private final ConnectContext session;
+        private List<String> userVariableDependencyNotInConnContext;
+
+        public CheckUserVariableDependencyVisitor(ConnectContext session,
+                                                  List<String> userVariableDependencyNotInConnContext) {
+            this.session = session;
+            this.userVariableDependencyNotInConnContext = userVariableDependencyNotInConnContext;
+        }
+
+        public Void visitUserVariableExpr(UserVariableExpr node, Scope context) {
+            UserVariable userVariable = session.getUserVariable(node.getName());
+            if (userVariable == null) {
+                userVariableDependencyNotInConnContext.add(node.getName());
+            }
+            return null;
+        }
+
+    }
+
+    public static List<String> analyzeUserVariableExprDependency(Expr expression, ConnectContext session) {
```

Comment:
can you write some comment for this function? What does this function do? And when is it used?